drv/stm32xx-i2c/src/lib.rs

```diff
@@ -148,33 +159,48 @@ pub enum ReadLength {
     Variable,
 }
 
+#[allow(clippy::upper_case_acronyms)]
 #[derive(Copy, Clone, Eq, PartialEq)]
+enum Register {
+    CR1,
+    CR2,
+    OAR1,
+    OAR2,
+    TIMINGR,
+    TIMEOUTR,
+    ISR,
+    PECR,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, counters::Count)]
 enum Trace {
-    WaitISR(u32),
-    WriteISR(u32),
-    WriteWaitISR(u32),
-    ReadISR(u32),
-    ReadWaitISR(u32),
-    RxISR(u32),
-    KonamiISR(u32),
-    Konami(I2cKonamiCode),
-    ResetISR(u32),
-    ResetCR2(u32),
-    AddrISR(u32),
+    Wait(Register, u32),
+    Write(Register, u32),
+    WriteWait(Register, u32),
+    Read(Register, u32),
+    ReadWait(Register, u32),
+    KonamiOperation(I2cKonamiCode),
+    Konami(Register, u32),
+    Reset(Register, u32),
+    Addr(Register, u32),
```

Comment:
if we want separate counts for each register we read/write to and we add a derived `Count` impl for `Register`, as I suggested above, we could add:
```suggestion
    Wait(#[count(children)] Register, u32),
    Write(#[count(children)] Register, u32),
    WriteWait(#[count(children)] Register, u32),
    Read(#[count(children)] Register, u32),
    ReadWait(#[count(children)] Register, u32),
    KonamiOperation(I2cKonamiCode),
    Konami(#[count(children)] Register, u32),
    Reset(#[count(children)] Register, u32),
    Addr(#[count(children)] Register, u32),
```

Reply:
I actually don't want to count all children there -- it's deliberate, even if/when we have `counted_ringbuf` here (which we can't right now per my other comment).